enterprise_access/apps/api/v1/views/content_assignments/assignments_admin.py

```diff
@@ -168,39 +171,71 @@ def list(self, request, *args, **kwargs):
 
     @extend_schema(
         tags=[CONTENT_ASSIGNMENT_ADMIN_CRUD_API_TAG],
-        summary='Cancel an assignment by UUID.',
+        summary='Cancel assignments by UUID.',
+        parameters=[LearnerContentAssignmentActionRequestSerializer],
         responses={
-            status.HTTP_200_OK: serializers.LearnerContentAssignmentAdminResponseSerializer,
+            status.HTTP_200_OK: None,
             status.HTTP_404_NOT_FOUND: None,
             status.HTTP_422_UNPROCESSABLE_ENTITY: None,
         },
     )
     @permission_required(CONTENT_ASSIGNMENT_ADMIN_WRITE_PERMISSION, fn=assignment_admin_permission_fn)
-    @action(detail=True, methods=['post'])
-    def cancel(self, request, *args, uuid=None, **kwargs):
+    @action(detail=False, methods=['post'])
+    def cancel(self, request, *args, **kwargs):
         """
-        Cancel a single ``LearnerContentAssignment`` record by uuid.
+        Cancel a list of ``LearnerContentAssignment`` records by uuid.
 
         Raises:
-            404 if the assignment was not found.
-            422 if the assignment is not cancelable.
+            404 if any of the assignments were not found
+            422 if any of the assignments threw an error
         """
+        serializer = LearnerContentAssignmentActionRequestSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        assignments = self.get_queryset().filter(
+            assignment_configuration__uuid=self.requested_assignment_configuration_uuid,
+            uuid__in=serializer.data['assignment_uuids'])
         try:
-            assignment_to_cancel = self.get_queryset().get(uuid=uuid)
-        except LearnerContentAssignment.DoesNotExist:
-            return Response(None, status=status.HTTP_404_NOT_FOUND)
-
-        # if the assignment is not cancelable, this is a no-op.
-        cancellation_info = assignments_api.cancel_assignments([assignment_to_cancel])
-
-        # If the response contains one element in the `cancelled` list, that is the one we sent, indicating succcess.
-        cancellation_succeeded = len(cancellation_info['cancelled']) == 1
-
-        if cancellation_succeeded:
-            # Serialize the assignment object obtained via get_queryset() instead of the one from the assignments_api.
-            # Only the former has the additional dynamic fields annotated, and those are required for serialization.
-            assignment_to_cancel.refresh_from_db()
-            response_serializer = serializers.LearnerContentAssignmentAdminResponseSerializer(assignment_to_cancel)
-            return Response(response_serializer.data, status=status.HTTP_200_OK)
-        else:
+            response = assignments_api.cancel_assignments(assignments)
+            if len(assignments) < len(request.data['assignment_uuids']):
+                return Response(status=status.HTTP_404_NOT_FOUND)
```

Comment:
This seems like a check we'd want to do before calling `cancel_assignments()`, right?  If this condition were true, it'd mean we couldn't find all of the assignments given the assignment uuids provided by the caller in our `filter()` above.

Reply:
No, because if we call it before then if any of the assignments are not found, it will just return before it hit the cancel rather than canceling the remaining ones.

Reply:
for posterity: we decided offline to leave the condition as-is, but change the response status to 422 here.